OrderService/OrderService.API/Controllers/OrderController.cs

```diff
@@ -0,0 +1,100 @@
+﻿using MediatR;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using OrderService.Application.Common.Dtos;
+using OrderService.Application.UseCases.OrderUseCases.Cancel;
+using OrderService.Application.UseCases.OrderUseCases.Create;
+using OrderService.Application.UseCases.OrderUseCases.CreateByBasket;
+using OrderService.Application.UseCases.OrderUseCases.GetByBuyerId;
+using OrderService.Application.UseCases.OrderUseCases.GetByFilters;
+using OrderService.Application.UseCases.OrderUseCases.GetBySellerId;
+using OrderService.Application.UseCases.OrderUseCases.UpdateStatus;
+
+namespace OrderService.API.Controllers
+{
+    [Authorize]
+    [Controller]
+    [Route("api/orders")]
+    public class OrderController(
+        IMediator mediator) : ControllerBase
+    {
+        [HttpGet("buyer")]
+        public async Task<IActionResult> GetOrdersByBuyerId(CancellationToken cancellationToken)
+        {
+            var buyerId = Guid.Parse(User.FindFirst("userId")!.Value);
+
+            var request = new GetOrdersByBuyerIdRequest(buyerId);
+
+            var orders = await mediator.Send(request, cancellationToken );
+
+
+            return Ok(orders);
+        }
+
+        [HttpGet("seller")]
+        public async Task<IActionResult> GetOrdersBySellerId(CancellationToken cancellationToken)
+        {
+            var sellerId = Guid.Parse(User.FindFirst("userId")!.Value);
+
+            var request = new GetOrdersBySellerIdRequest(sellerId);
+
+            var orders = await mediator.Send(request, cancellationToken);
+
+            return Ok(orders);
+        }
+
+        [HttpGet]
+        public async Task<IActionResult> GetOrdersByFilters([FromQuery] GetOrdersByFiltersRequest request, CancellationToken cancellationToken)
+        {
+            var orders = await mediator.Send(request, cancellationToken);
+
+            return Ok(orders);
+        }
+
+        [HttpPatch("{orderId}")]
+        public async Task<IActionResult> UpdateOrderStatus(Guid orderId, UpdateOrderStatusRequestDto dto, CancellationToken cancellationToken)
```

Comment:
please specify from where UpdateOrderStatusRequestDto is binding (frombody, fromquery or etc.)